common/flatpak-transaction.c

```diff
@@ -1982,6 +2078,18 @@ load_deployed_metadata (FlatpakTransaction *self, const char *ref)
   return g_bytes_new_take (g_steal_pointer (&metadata_contents), metadata_contents_length + 1);
 }
 
+static gboolean
+check_skip_eol (FlatpakTransaction          *self,
+                FlatpakTransactionOperation *op)
+{
+  if (op->skip || (!op->eol && !op->eol_rebase) ||
+      op->kind == FLATPAK_TRANSACTION_OPERATION_UNINSTALL)
+    return FALSE;
+
+  g_signal_emit (self, signals[END_OF_LIFED_PRE], 0, op->remote, op->ref, op->eol, op->eol_rebase, &op->skip);
```

Comment:
I don't like how this calls out to some random code which can add ops to the transaction. In fact, new operations are prepended in flatpak_transaction_add_op(), so it will only happen in the resolve ops that happen after the added deps in flatpak_transaction_real_run().

I think we need to set a op_added boolean in flatpak_transaction_add_op(), and then loop around resolve_op() if it was set.